lib/Target/CBackend/CBackend.cpp

```diff
@@ -1695,13 +1707,26 @@ void CWriter::writeOperandInternal(Value *Operand,
 
 void CWriter::writeOperand(Value *Operand, enum OperandContext Context) {
   bool isAddressImplicit = isAddressExposed(Operand);
-  if (isAddressImplicit)
-    Out << "(&"; // Global variables are referenced as their addresses by llvm
+  bool isEmpty = false;
+  if (isAddressImplicit) {
+    // We can't directly declare a zero-sized variable in C, so
+    // printTypeNameForAddressableValue uses a single-byte type instead.
+    // We fix up the pointer type here.
+    if (isEmptyType(Operand->getType()->getPointerElementType())) {
+      Out << "(void*)(";
```

Comment:
I think I meant `((void*)` because otherwise this extra layer of parentheses is useless.